packages/loopring_v3/contracts/impl/libtransactions/AccountTransferTransaction.sol

```diff
@@ -142,6 +128,23 @@ library OwnerChangeTransaction
             // If the account has a wallet, use it to recover the account
             require(accountTransfer.walletHash != 0, "ACCOUNT_HAS_NO_WALLET");
 
+            // Verify authorization from the new owner
```

Comment:
I don't think we need to get authorization from the new owner, otherwise, the user experience will be bad. Even in hebao, we don't need a signature from the new owner.

Reply:
I removed the requirement when the old owner can still sign, but kept this one because someone needs to sign something. Basically we have the same problem as with our smart wallet: who can sign meta txs for an account. The new owner is required to sign because he authorizes:
- The fee payment (using funds from the account being recovered, so only the old or the new owner can do that)
- Which data is used to recover the account

I guess we could let another account pay for the recovery, but I think that makes things a bit more complicated, and in any case someone needs to sign the transaction so why not the new owner and keep things simple?

Reply:
We can/should always assume that if the current owner configs his wallet to have guardians, then a majority of the guardian's approval does authorize the operator to charge the recovery fee out of this account. I don't think the new owner needs to sign. Most likely one of his guardians is another address of his.

"Which data is used to recover the account"
I think we can simply reset the wallet data after the recovery, let the new owner to config his wallet later.

Reply:
That's a good point, but they'd have approve the fee payment as well which they currently do not, but that makes sense.